yaml_edit/lib/src/errors.dart

```diff
@@ -76,3 +76,36 @@ class AliasError extends UnsupportedError {
             'by this library.\n\n'
             '${anchor.span.message('The alias was first defined here.')}');
 }
+
+/// Error thrown when an assertion about the YAML fails. Extends
+/// [AssertionError] to override the [toString] method for pretty printing.
+class YamlAssertionError extends AssertionError {
+  YamlAssertionError(message) : super(message);
+
+  @override
+  String toString() {
+    if (message != null) {
+      return 'Assertion failed: ${message}';
+    }
+    return 'Assertion failed';
+  }
+}
+
+/// Throws an [AssertionError] with the given [message], and format
+/// [oldYaml] and [newYaml] for information.
+@alwaysThrows
+void createAssertionError(String message, String oldYaml, String newYaml) {
+  throw YamlAssertionError('''
```

Comment:
I propose we `return _YamlAssertionError` and that we always throw when calling this.. That way we get a better stack trace.

For error correlation services the top of the stack trace if often important.